test/hummingbot/strategy/cross_exchange_market_making/test_cross_exchange_market_making_gateway.py

```diff
@@ -0,0 +1,1017 @@
+                OrderType.LIMIT,
+                ask_order.trading_pair,
+                ask_order.quantity,
+                ask_order.price,
+                ask_order.client_order_id,
+                ask_order.creation_timestamp * 1e-6,
+            )
+        )
+
+        self.taker_market.set_prices(
+            self.taker_trading_pairs[0],
+            True,
+            1.01
+        )
+        self.taker_market.set_prices(
+            self.taker_trading_pairs[0],
+            False,
+            0.99
+        )
+
+        self.clock.backtest_til(self.start_timestamp + 10)
+        await asyncio.sleep(0.5)
+        self.clock.backtest_til(self.start_timestamp + 11)
+        await asyncio.sleep(0.5)
+
+        self.assertEqual(2, len(self.cancel_order_logger.event_log))
+        self.assertEqual(1, len(self.strategy.active_maker_bids))
+        self.assertEqual(1, len(self.strategy.active_maker_asks))
+
+        bid_order = self.strategy.active_maker_bids[0][1]
+        ask_order = self.strategy.active_maker_asks[0][1]
+        self.assertEqual(Decimal("0.98507"), bid_order.price)
+        self.assertEqual(Decimal("1.0151"), ask_order.price)
+
+        self.clock.backtest_til(self.start_timestamp + 20)
+        await asyncio.sleep(0.5)
+        self.simulate_limit_order_fill(self.maker_market, bid_order)
+        self.simulate_limit_order_fill(self.maker_market, ask_order)
+
+        self.clock.backtest_til(self.start_timestamp + 25)
+        await asyncio.sleep(0.5)
+        self.clock.backtest_til(self.start_timestamp + 30)
+        await asyncio.sleep(0.5)
+
+        # Order fills not emitted by the gateway for now
+        # fill_events: List[OrderFilledEvent] = self.taker_order_fill_logger.event_log
+
+        # bid_hedges: List[OrderFilledEvent] = [evt for evt in fill_events if evt.trade_type is TradeType.SELL]
+        # ask_hedges: List[OrderFilledEvent] = [evt for evt in fill_events if evt.trade_type is TradeType.BUY]
+        # self.assertEqual(1, len(bid_hedges))
+        # self.assertEqual(1, len(ask_hedges))
+        # self.assertGreater(
+        #    self.maker_market.get_balance(self.maker_trading_pairs[2]) + self.taker_market.get_balance(self.taker_trading_pairs[2]),
+        #    Decimal("10"),
+        # )
+        # Order fills not emitted by the gateway for now
+        # self.assertEqual(2, len(self.taker_order_fill_logger.event_log))
+        # taker_fill1: OrderFilledEvent = self.taker_order_fill_logger.event_log[0]
+        # self.assertEqual(TradeType.SELL, taker_fill1.trade_type)
+        # self.assertAlmostEqual(Decimal("0.9895"), taker_fill1.price)
+        # self.assertAlmostEqual(Decimal("3.0"), taker_fill1.amount)
+        # taker_fill2: OrderFilledEvent = self.taker_order_fill_logger.event_log[1]
+        # self.assertEqual(TradeType.BUY, taker_fill2.trade_type)
+        # self.assertAlmostEqual(Decimal("1.0105"), taker_fill2.price)
+        # self.assertAlmostEqual(Decimal("3.0"), taker_fill2.amount)
```

Comment:
Is this still required?

Reply:
I kept it there in case once the gateway starts emitting the even as other connectors. I can delete it, it was just for future convenience and to clarify why the rest of the test is missing.